entrypoint.sh

```diff
@@ -13,58 +13,61 @@ if [ -z "$PR_NUMBER" ]; then
   exit 1
 fi
 
-GITHUB_REPOSITORY_NAME=${GITHUB_REPOSITORY#$GITHUB_REPOSITORY_OWNER/}
+# Check the flyctl version
+flyctl version
+
+REPO_NAME=$(jq -r .repository.name /github/workflow/event.json)
 EVENT_TYPE=$(jq -r .action /github/workflow/event.json)
 
-# Default the Fly app name to pr-{number}-{repo_owner}-{repo_name}
-app="${INPUT_NAME:-pr-$PR_NUMBER-$GITHUB_REPOSITORY_OWNER-$GITHUB_REPOSITORY_NAME}"
-# Change underscores to hyphens.
-app="${app//_/-}"
-region="${INPUT_REGION:-${FLY_REGION:-iad}}"
+# Default the Fly app name to {repo_name}-pr-{pr_number}
+app="${INPUT_NAME:-$REPO_NAME-pr-$PR_NUMBER}"
+# # Default the Fly app name to {repo_name}-pr-{pr_number}-postgres
+postgres_app="${INPUT_POSTGRES:-$REPO_NAME-pr-$PR_NUMBER-postgres}"
+region="${INPUT_REGION:-${FLY_REGION:-ord}}"
 org="${INPUT_ORG:-${FLY_ORG:-personal}}"
-image="$INPUT_IMAGE"
-config="${INPUT_CONFIG:-fly.toml}"
+dockerfile="$INPUT_DOCKERFILE"
+
+# only wait for the deploy to complete if the user has requested the wait option
+# otherwise detach so the GitHub action doesn't run as long
+if [ "$INPUT_WAIT" = "true" ]; then
+  detach=""
+else
+  detach="--detach"
+fi
 
 if ! echo "$app" | grep "$PR_NUMBER"; then
   echo "For safety, this action requires the app's name to contain the PR number."
   exit 1
 fi
 
-# PR was closed - remove the Fly app if one exists and exit.
+# If PR is closed or merged, the app will be deleted
 if [ "$EVENT_TYPE" = "closed" ]; then
   flyctl apps destroy "$app" -y || true
+
+  message="Review app deleted."
+  echo "message=$message" >> $GITHUB_OUTPUT
   exit 0
 fi
 
 # Deploy the Fly app, creating it first if needed.
 if ! flyctl status --app "$app"; then
-  # Backup the original config file since 'flyctl launch' messes up the [build.args] section
-  cp "$config" "$config.bak"
-  flyctl launch --no-deploy --copy-config --name "$app" --image "$image" --region "$region" --org "$org"
-  # Restore the original config file
-  cp "$config.bak" "$config"
-fi
-if [ -n "$INPUT_SECRETS" ]; then
-  echo $INPUT_SECRETS | tr " " "\n" | flyctl secrets import --app "$app"
-fi
+  flyctl launch --no-deploy --copy-config --name "$app" --dockerfile "$dockerfile" --region "$region" --org "$org"
 
-# Attach postgres cluster to the app if specified.
-if [ -n "$INPUT_POSTGRES" ]; then
-  flyctl postgres attach "$INPUT_POSTGRES" --app "$app" || true
-fi
+  # Attach postgres cluster and set the DATABASE_URL
+  flyctl postgres attach "$postgres_app" --app "$app"
+  flyctl deploy $detach --app "$app" --region "$region" --strategy immediate --remote-only
```

Comment:
what if I have a different fly.toml for CI than for prod?

Reply:
Correct me if i am wrong, `fly.toml` is used for prod by default right ? So when these scripts are executed it looks for an exiting `fly.toml` file. However if we have one more toml file for CI then i don't think it will use that file at all when the scripts are executed.